### PR TITLE
jsmith fix attempt 2 for katcp_fpga/program()

### PR DESCRIPTION
**Pull request by james-smith-za**
Merges in https://github.com/ska-sa/casperfpga/pull/10 and my own fixes. Works perfectly for me so far.

### CODE REVIEW

File: src/katcp_fpga.py [hunk 1]
```diff
@@ -218,6 +226,7 @@ def program(self, filename=None):
             elif filename != self.system_info['program_filename']:
                 LOGGER.error('Programming filename %s, configured programming filename %s'
                              % (filename, self.system_info['program_filename']))
+                # This doesn't seem as though it should really be an error...
```
Comment (brickZA) [inline src/katcp_fpga.py:232]: Seems to be pointing out some kind of inconsistency. @paulprozesky ?

File: src/katcp_fpga.py [hunk 2]
```diff
@@ -65,7 +65,15 @@ def connect(self, timeout=1):
         :return:
         """
         if not self.is_connected():
-            self.start(daemon=True)
+            # Implement backward / forwards compabitlity for change in daemonization APIs
```
Comment (brickZA) [inline src/katcp_fpga.py]: Weird, I thought we merged this fix a while ago :)  Small spelling mistake: compabitlity -> compatibility

File: src/katcp_fpga.py (hunk 1 above)
Comment (paulprozesky) [inline src/katcp_fpga.py:232]: Well, the idea was that you could set the self.system_info['program_filename'] variable and then just say self.program(). So if you'd set it, and then passed in a different filename, that wouldn't be correct.

**Comment by paulprozesky** [comment]
Otherwise the changes look good. Merge away.

**Comment by brickZA** [comment]
@james-smith-za Could you merge with ska-sa's devel branch first? It does not currenly apply cleanly.

**Comment by james-smith-za** [comment]
I'm pretty sure I did... let me try again.

**Comment by james-smith-za** [comment]
Ok, merged devel branch. Try again?
